services/submission.js

```diff
@@ -229,8 +170,92 @@ class Submission {
             }    
         }
     }
+    /**
+     * API: submissionAction
+     * @param {*} params 
+     * @param {*} context 
+     * @returns updated submission
+     */
+    async submissionAction(params, context){
+        verifySession(context)
+            .verifyInitialized();
+        const userInfo = context.userInfo;
+        const submissionID = params?.submissionID;
+        const action = params?.action;
+        //verify submission action
+        const verifier = verifySubmissionAction(submissionID, action);
+        //verify if a submission can be find by submissionID.
+        let submission = await verifier.exists(this.submissionCollection);
```

Comment:
I think we should not pass this.submissionCollection. The verifier should be separate, focusing only on verification.
const submission = await this.submissionCollection(id);
await verifier.exists(this.submissionCollection);

Reply:
I will consider, but my implement is a one stone server birds solution that encapsulate logics to verify the submissionId and check if submission can be found.